core/test/nuxeo-audit-page-provider.test.js

```diff
@@ -0,0 +1,236 @@
+const user = 'Administrator';
+const from = moment().format('YYYY-MM-DD');
+const to = moment()
+  .add(5, 'days')
+  .format('YYYY-MM-DD');
+const requestParameters = {
+  startEventDate: from,
+  endEventDate: to,
+  category: eventDocumentCategory,
+  eventId: events,
+  principalName: user,
+};
+const providerParameters = {
+  startDate: from,
+  endDate: to,
+  eventCategory: eventDocumentCategory,
+  eventIds: events,
+  principalName: user,
+};
+const providerSort = {
+  eventDate: 'asc',
+  category: 'desc',
+  eventId: 'desc',
+  principalName: 'asc',
+};
+let server;
+async function getProvider(params = {}, docId = documentId, sort = {}) {
+  const provider = await fixture(
+    html`
+      <nuxeo-audit-page-provider
+        id="provider"
+        doc-Id="${docId}"
+        params="${JSON.stringify(params)}"
+        sort="${JSON.stringify(sort)}"
+        page-size="40"
+      ></nuxeo-audit-page-provider>
+    `,
+  );
+  return provider;
+}
+
+function checkRequestQueryParams(qParams = {}) {
+  const expectedQueryParam = { currentPageIndex: 0, pageSize: 40, ...qParams };
+  const { queryParams } = server.getLastRequest('GET', documentHistoryPath);
+  expect(expectedQueryParam).to.eql(queryParams);
+}
+
+function mockDocHistoryResponse() {
+  server.respondWith('GET', documentHistoryPath, { 'entity-type': 'logEntries', entries: [] });
+}
+
+function mockAuditResponse() {
+  server.respondWith('POST', adminAuditPath, { 'entity-type': 'logEntries', entries: [] });
+}
+
+function checkRequestBodyParams(requestParams = {}) {
+  // build request parameters and sort (optional) from the body
+  const { body } = server.getLastRequest('POST', adminAuditPath);
+  const { currentPageIndex, pageSize, providerName, namedQueryParams, sortBy, sortOrder } = body.params;
+  const flatedQueryParams = Object.assign({}, namedQueryParams);
+  // eventIds is a stringified array, it should be parse to a JS array
+  if ('eventIds' in flatedQueryParams) {
+    flatedQueryParams.eventIds = JSON.parse(flatedQueryParams.eventIds);
+  }
+  const bodyParams = {
+    currentPageIndex,
+    pageSize,
+    providerName,
+    ...(sortBy && { sortBy }),
+    ...(sortOrder && { sortOrder }),
+    ...flatedQueryParams,
+  };
+
+  // build the expected parameters (add the commons properties)
+  const expectedRequestParam = { currentPageIndex: 0, pageSize: 40, providerName: 'EVENTS_VIEW', ...requestParams };
+
+  expect(expectedRequestParam).to.eql(bodyParams);
+}
+
+function checkResponse(response) {
+  expect(response).to.be.an('object');
+  expect(response).to.have.property('entity-type');
+  expect(response).to.have.property('entries');
+  expect(response.entries).to.be.an('array').that.is.empty;
+}
+
+suite('nuxeo-audit-page-provider', () => {
+  setup(() => {
+    server = fakeServer.create();
+  });
+
+  teardown(() => {
+    server.restore();
+  });
+
+  suite('Fetch Document History', () => {
+    test('Should get history without parameters', async () => {
```

Comment:
If possible we should try to keep the "convention" started for tests' names: "**Should** X **when** Y".
In this case, something like `Should fetch audit history when no parameters are used` would make sense?